### ISSUE TITLE
Bumpy Charcoal Squid - `RFPSimpleStrategy::_distribute` function: rejected milestones can be executed

### ISSUE THREAD

**Issue by sherlock-admin2** [sherlock-audit/2023-09-Gitcoin-judging#727]
Bumpy Charcoal Squid

high

# `RFPSimpleStrategy::_distribute` function: rejected milestones can be executed
## Summary

`RFPSimpleStrategy::_distribute` function: rejected milestones can be executed

## Vulnerability Detail

- In `RFPSimpleStrategy` strategy contract: the pool manager can set a list of milestones of payments to be distributed to the selected accepted recipient (`acceptedRecipientId`) later, and this operation can be done only once via `setMilestones` function.

- The `acceptedRecipientId` can submit a milestone: in order to get the next milestone payment when this milestone is distributed; and this sets the milestone status from `None` to `Pending`.

- Then the pool manager can either reject the submitted milestone by calling `rejectMilestone` function where it sets the status of the milestone from `Pending` to `Rejected`; or can leave the status as it is (`Pending`).

- But when the `_duistribute` function is called for the nextMilestone; it doesn't check if the status of the milestone is **Rejected**.

## Impact

So this will result in distributing the **Rejected** milestone payment for the accepted recipient.

## Code Snippet

[RFPSimpleStrategy::submitUpcomingMilestone function](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L253-L271)

```solidity
    function submitUpcomingMilestone(Metadata calldata _metadata) external {
        // Check if the 'msg.sender' is the 'acceptedRecipientId' and is a member of the 'Profile'
        if (acceptedRecipientId != msg.sender && !_isProfileMember(acceptedRecipientId, msg.sender)) {
            revert UNAUTHORIZED();
        }

        // Check if the upcoming milestone is in fact upcoming
        if (upcomingMilestone >= milestones.length) revert INVALID_MILESTONE();

        // Get the milestone and update the metadata and status
        Milestone storage milestone = milestones[upcomingMilestone];
        milestone.metadata = _metadata;

        // Set the milestone status to 'Pending' to indicate that the milestone is submitted
        milestone.milestoneStatus = Status.Pending;

        // Emit event for the milestone
        emit MilstoneSubmitted(upcomingMilestone);
    }
```

[RFPSimpleStrategy::rejectMilestone function](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L283-L290)

```solidity
    function rejectMilestone(uint256 _milestoneId) external onlyPoolManager(msg.sender) {
        // Check if the milestone is already accepted
        if (milestones[_milestoneId].milestoneStatus == Status.Accepted) revert MILESTONE_ALREADY_ACCEPTED();

        milestones[_milestoneId].milestoneStatus = Status.Rejected;

        emit MilestoneStatusChanged(_milestoneId, milestones[_milestoneId].milestoneStatus);
    }
```

[RFPSimpleStrategy::\_distribute function](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L417-L450)

```solidity
function _distribute(address[] memory, bytes memory, address _sender)
        internal
        virtual
        override
        onlyInactivePool
        onlyPoolManager(_sender)
    {
        // check to make sure there is a pending milestone
        if (upcomingMilestone >= milestones.length) revert INVALID_MILESTONE();

        IAllo.Pool memory pool = allo.getPool(poolId);
        Milestone storage milestone = milestones[upcomingMilestone];
        Recipient memory recipient = _recipients[acceptedRecipientId];

        // make sure has enough funds to distribute based on the proposal bid
        if (recipient.proposalBid > poolAmount) revert NOT_ENOUGH_FUNDS();

        // Calculate the amount to be distributed for the milestone
        uint256 amount = (recipient.proposalBid * milestone.amountPercentage) / 1e18;

        // Get the pool, subtract the amount and transfer to the recipient
        poolAmount -= amount;
        _transferAmount(pool.token, recipient.recipientAddress, amount);

        // Set the milestone status to 'Accepted'
        milestone.milestoneStatus = Status.Accepted;

        // Increment the upcoming milestone
        upcomingMilestone++;

        // Emit events for the milestone and the distribution
        emit MilestoneStatusChanged(upcomingMilestone, Status.Accepted);
        emit Distributed(acceptedRecipientId, recipient.recipientAddress, amount, _sender);
    }
```

## Tool used

Manual Review

## Recommendation

Check the milestone status before distribution; if it's set to `Rejected`, then `_distribute` function won't execute this milestone and moves to the next one for execution, and add a mechanism to enable executing the `Rejected` milestones again when their statuses are updated (impelemnt another mechanism to update the milestone statuses by the pool manager).

**Comment by DevHals** [comment]
Escalate

I think this issue needs a second look!

first of all, let me brief the distribution mechanism in the `RFPSimpleStrategy`:

1. the pool manager sets the milestones that are going to be distributed
2. then the milestones are ditributed to the accepted recipient **sequentially**; meaning that the first milestone is distributed then the second then the third... and this is ensured by increasing the pointer (index) of the next milestone after each distribution:
   [here](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L428)

```solidity
Milestone storage milestone = milestones[upcomingMilestone];
```

and [here](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L445)

```solidity
upcomingMilestone++;
```

and the `RFPSimpleStrategy` type has a functionality for the accepted recipient to submit the milestone (via  [submitUpcomingMilestone](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L253-L272)) to be either accepted or rejected by the pool manger (rejecting via [rejectMilestone](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L283-L290)); if accepted then this milestone should be distributed to the accepted recipient; and if it is rejected then it **shouldn't** be distributed to the accepted recipient.

and as you can see the problem here:

**there's no check on the milestone status if it's rejected before distributing it; and this has nothing to do with the manager being trusted or not, why? here's why:**

because the current distribution mechanism will distribute the milestones **sequentially**, and the manager can't control which milestone to distribute, all he can do is calling the `distribute` function to initiate the milestone distribution; and since there's no check on the milestone's status; then all milestones will be distributed even the **rejected** ones!

I have added a suggestion to fix this issue in my report:

> ## Recommendation
>
> Check the milestone status before distribution; if it's set to `Rejected`, then `_distribute` function won't execute this milestone and moves to the next one for execution, and add a mechanism to enable executing the `Rejected` milestones again when their statuses are updated (impelemnt another mechanism to update the milestone statuses by the pool manager).

**Comment by sherlock-admin2** [comment]
> Escalate
> 
> I think this issue needs a second look!
> 
> first of all, let me brief the distribution mechanism in the `RFPSimpleStrategy`:
> 
> 1. the pool manager sets the milestones that are going to be distributed
> 2. then the milestones are ditributed to the accepted recipient **sequentially**; meaning that the first milestone is distributed then the second then the third... and this is ensured by increasing the pointer (index) of the next milestone after each distribution:
>    [here](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L428)
> 
> ```solidity
> Milestone storage milestone = milestones[upcomingMilestone];
> ```
> 
> and [here](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L445)
> 
> ```solidity
> upcomingMilestone++;
> ```
> 
> and the `RFPSimpleStrategy` type has a functionality for the accepted recipient to submit the milestone (via  [submitUpcomingMilestone](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L253-L272)) to be either accepted or rejected by the pool manger (rejecting via [rejectMilestone](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/6430c8004017e96ae2f5aac365bdefd0b6eeea72/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol#L283-L290)); if accepted then this milestone should be distributed to the accepted recipient; and if it is rejected then it **shouldn't** be distributed to the accepted recipient.
> 
> and as you can see the problem here:
> 
> **there's no check on the milestone status if it's rejected before distributing it; and this has nothing to do with the manager being trusted or not, why? here's why:**
> 
> because the current distribution mechanism will distribute the milestones **sequentially**, and the manager can't control which milestone to distribute, all he can do is calling the `distribute` function to initiate the milestone distribution; and since there's no check on the milestone's status; then all milestones will be distributed even the **rejected** ones!
> 
> I have added a suggestion to fix this issue in my report:
> 
> > ## Recommendation
> >
> > Check the milestone status before distribution; if it's set to `Rejected`, then `_distribute` function won't execute this milestone and moves to the next one for execution, and add a mechanism to enable executing the `Rejected` milestones again when their statuses are updated (impelemnt another mechanism to update the milestone statuses by the pool manager).

You've created a valid escalation!

To remove the escalation from consideration: Delete your comment.

You may delete or edit your escalation comment anytime before the 48-hour escalation window closes. After that, the escalation becomes final.

**Comment by DevHals** [comment]
Hi @thelostone-mc,
[the implemented fix of this issue](https://github.com/allo-protocol/allo-v2/commit/094e8b2426e213053ba0d0d5dbfa41fc26271a37) will not solve it; but will worsen it:
 because if there's any rejected milestone then the distribution will be disabled since the milestones distribution is **sequential** and can't be controlled by determining which milestone to distribute first, and there's no function to update the milestone status from rejected to pending,,

**Comment by DevHals** [comment]
Hi @thelostone-mc,
[the implemented fix of this issue](https://github.com/allo-protocol/allo-v2/commit/094e8b2426e213053ba0d0d5dbfa41fc26271a37) will not solve it; but will add an extra issue:
 because if there's any rejected milestone then the distribution will be disabled since the milestones distribution is **sequential** and can't be controlled by determining which milestone to distribute first, and there's no function to update the milestone status from rejected to pending,,

**Comment by neeksec** [comment]
Suggest to keep invalid.

> all he can do is calling the distribute function to initiate the milestone distribution; and since there's no check on the milestone's status; then all milestones will be distributed even the rejected ones!

`_distribute` only distribute the `upcomingMilestone`. Not all milestones.

**Comment by DevHals** [comment]
@neeksec 
your comment seems to be on something else apart from the explained issue:

Could you please re-read my escalation and my comment on the implemented [fix](https://github.com/allo-protocol/allo-v2/commit/094e8b2426e213053ba0d0d5dbfa41fc26271a37) by the team to get what exactly is going on?

the `upcomingMilestone` is incremented after each distribution, the manager calls `distribute` and the `upcomingMilestone` is distributed,, with **the current fix implemented by the team**; the `distribute` function will be blocked if any of the milestones is rejected and will not be able to distribute any milestone after the rejected one,,

**Comment by Evert0x** [comment]
@neeksec `upcomingMilestone` is incremented with `++`, that means all milestones are rewarded sequentially, even the rejected ones, right?

**Comment by neeksec** [comment]
@Evert0x If the upcoming milestone is rejected, manager should not call distribute(operational error if call). The `upcomingMilestone` won't be incremented. There will never be rejected milestones in the past. So distribute only distribute the accepted ones.

**Comment by DevHals** [comment]
@neeksec 

The protocol team confirmed this issue  #510, and introduced a fix for it,

as per your justification; the manager **should never** call `distribute` function again if he rejected the next milestone,right?
by assuming that the manager **will not call distribute** function again because he rejected the next milestone; wouldn't this block the distribution and the allocated funds being stuck in the contract? also bear in mind that there's multiple managers for each pool not only one and there must be a sort of off-chain communication between them to coordinate to not call this function again, is this reasonable?

I have updated the `RFPSimpleStrategy` contract as per the [fix](https://github.com/allo-protocol/allo-v2/commit/094e8b2426e213053ba0d0d5dbfa41fc26271a37) and set-up a test to demonstrate the current design of accepting and rejecting milestones will block the distribution of the allocated funds which leads to funds being stuck as **there's no function for the manager to update the status of the rejected milestones** :

While setting the test for the fix; I have discovered another vulnerability where the accepted recipient can bypass the rejection by calling the `submitUpcomingMilestone` function on the rejected milestone **which should be prevented**.

1. The updated `RFPSimpleStrategy._distribute` function:

   ```diff
   function _distribute(address[] memory, bytes memory, address _sender)
           internal
           virtual
           override
           onlyInactivePool
           onlyPoolManager(_sender)
       {
           // check to make sure there is a pending milestone
           if (upcomingMilestone >= milestones.length) revert INVALID_MILESTONE();

           IAllo.Pool memory pool = allo.getPool(poolId);
           Milestone storage milestone = milestones[upcomingMilestone];
           Recipient memory recipient = _recipients[acceptedRecipientId];

           // make sure has enough funds to distribute based on the proposal bid
           if (recipient.proposalBid > poolAmount) revert NOT_ENOUGH_FUNDS();

   +        // Check if the milestone is pending
   +       if (milestone.milestoneStatus != Status.Pending)
   +           revert INVALID_MILESTONE();

           // Calculate the amount to be distributed for the milestone
           uint256 amount = (recipient.proposalBid * milestone.amountPercentage) / 1e18;

           // Get the pool, subtract the amount and transfer to the recipient
           poolAmount -= amount;
           _transferAmount(pool.token, recipient.recipientAddress, amount);

           // Set the milestone status to 'Accepted'
           milestone.milestoneStatus = Status.Accepted;

           // Increment the upcoming milestone
           upcomingMilestone++;

           // Emit events for the milestone and the distribution
           emit MilestoneStatusChanged(upcomingMilestone, Status.Accepted);
           emit Distributed(acceptedRecipientId, recipient.recipientAddress, amount, _sender);
       }
   ```

2. Foundry PoC: add the following test `test_Rejecting_Milestone_Will_Block_Distributions` in `allo-v2/test/foundry/strategies/RFPSimpleStrategy.t.sol` :

   ```solidity
       function test_Rejecting_Milestone_Will_Block_Distributions() public {
           //Note : this test is copied from "test_rejectMilestone" and updated to illustrate the introduced vulnerability of rejecting milestones:

           //0. first the pool is funded :
           vm.deal(pool_admin(), 1e19);
           vm.prank(pool_admin());
           allo().fundPool{value: 1e19}(poolId, 1e19);

           //1. the accepted recipient submitted the upcoming milestone:
           __register_setMilestones_allocate_submitUpcomingMilestone();
           RFPSimpleStrategy.Milestone memory milestone = strategy.getMilestone(0);
           assertEq(
               uint8(milestone.milestoneStatus),
               uint8(IStrategy.Status.Pending)
           );

           //2. the manager "rejects" the submitted (upcoming milestone):
           vm.expectEmit();
           emit MilestoneStatusChanged(0, IStrategy.Status.Rejected);
           vm.prank(pool_admin());
           strategy.rejectMilestone(0);
           milestone = strategy.getMilestone(0);
           assertEq(
               uint8(milestone.milestoneStatus),
               uint8(IStrategy.Status.Rejected)
           );

           //3. the manager calls distribute but it will revert because of the fix where a check is made on the status of the milestone before distributing it:
           vm.prank(address(allo()));
           vm.expectRevert();
           strategy.distribute(new address[](0), "", pool_admin());

           //4. BUT the  accepted recipient can bypass the rejected milestone by the manager and resets it again!
           vm.expectEmit();
           emit MilstoneSubmitted(0);
           vm.prank(recipient());
           strategy.submitUpcomingMilestone(
               Metadata({protocol: 0, pointer: "metadata"})
           );

           //5. and if any of the managers calls the distribute function; then the previously rejected milestone that has been re-set again by the accepted recipient will be executed (as the milestones are inceremented sequentially); and this should be avoided!
           vm.prank(address(allo()));
           strategy.distribute(new address[](0), "", pool_admin());
       }
   ```

3. Test result:

   ```bash
   $ forge test --match-test test_Rejecting_Milestone_Will_Block_Distributions
   Running 1 test for test/foundry/strategies/RFPSimpleStrategy.t.sol:RFPSimpleStrategyTest
   [PASS] test_Rejecting_Milestone_Will_Block_Distributions() (gas: 607634)
   Test result: ok. 1 passed; 0 failed; 0 skipped; finished in 9.72ms
   Ran 1 test suites: 1 tests passed, 0 failed, 0 skipped (1 total tests)
   ```

**Comment by Evert0x** [comment]
>  because he rejected the next milestone; wouldn't this block the distribution and the allocated funds being stuck in the contract? 

My understanding is the answer to this question is yes. 

@neeksec

**Comment by neeksec** [comment]
> because he rejected the next milestone; wouldn't this block the distribution and the allocated funds being stuck in the contract?

I don't think so. If he rejected the next milestone, he should not distribute and should wait for recepient to re-submit the milestone. Re-submiting milestone will rewrite the upcoming milestone status to pending. The `upcomingMilestone` pointer will not advance when re-submit. Manager only distribute when he saw a satisfying upcoming milestone. The fund can only be locked if manager distributes rejected milestone. This can be blocked by the patch, but it's mainly manager's operational fault to distribute rejected milestone in my opinion.

**Comment by DevHals** [comment]
> > because he rejected the next milestone; wouldn't this block the distribution and the allocated funds being stuck in the contract?
> 
> I don't think so. If he rejected the next milestone, he should not distribute and should wait for recepient to re-submit the milestone. Re-submiting milestone will rewrite the upcoming milestone status to pending. The `upcomingMilestone` pointer will not advance when re-submit. Manager only distribute when he saw a satisfying upcoming milestone. The fund can only be locked if manager distributes rejected milestone. This can be blocked by the patch, but it's mainly manager's operational fault to distribute rejected milestone in my opinion.

So basically you  used the other discovered vulnerability (where the acceptedRecipient re-sets the status of the rejected milestone to `Pending`) as a work around solution for the described vulnerability?! 
while **re-setting** the status of rejected milestones should be done by the manager only!

**as per your justification: the manager should refrain from calling the `distribute` function because he has rejected the upcoming milestone,,** 
What if the manager didn't wish to alter the status of the upcoming rejected milestone for valid reasons (such as the current  recipient is ineligible to receive this specific milestone) but he wants to distribute the next milestone that comes after the rejected one for this recipient? he will not be able to do so since the distribution is blocked by the rejected one (and the pointer will not be updated unless this rejected one is executed)!

Sorry, but there's a very obvious flaw in the milestones distribution/rejection mechanism and none of the provided reasons to invalidate the issue seems to be convincing,,

A clearer perspective on the issue can be obtained by referring to the recommended solution:

> ## Recommendation 
> Check the milestone status before distribution; if it's set to `Rejected`, then `_distribute` function won't execute this milestone and moves to the next one for execution, and add a mechanism to enable executing the `Rejected` milestones again when their statuses are updated (impelemnt another mechanism to update the milestone statuses by the pool manager).

**Comment by Evert0x** [comment]
@DevHals 

> So basically you used the other discovered vulnerability 

Can you link to this vulnerability?

**Comment by neeksec** [comment]
> So basically you used the other discovered vulnerability (where the acceptedRecipient re-sets the status of the rejected milestone to Pending) as a work around solution for the described vulnerability?!
> while re-setting the status of rejected milestones should be done by the manager only!

No. No workaround needs to be used.
Recipient should redo his work the milestone requires and call `submitUpcomingMilestone` to re-submit if his submitted milestone had been rejected. This will reset the milestone status to Pending and made the redo work ready to be re-reviewed by manager.

This is not a vulnerability and is the intended workflow.

**Comment by DevHals** [comment]
> @DevHals
> 
> > So basically you used the other discovered vulnerability
> 
> Can you link to this vulnerability?

@Evert0x 
it's caused by the current fix,  where the `_distribute` function checks if the status is `== Pending` to proceed, while the recipient can still call the `submitUpcomingMilestone` again to change the status of the milestone from `Rejected` to `Pending` (this should be done by the manager); then when the manager calls the `_distribute`; this previously rejected milestone by the manager that has been reset by the recipient will be executed,,,and assuming the manager should refrain from calling the distributions again until the upcoming milestone status is updated is not a solution,(also each pool has multiple managers not only one, so any manager can mistakenly distribute this milestone),,,
I have updated the `_distribute` function as per the fix and wrote a PoC to demonstrate it in my  [previous reply](https://github.com/sherlock-audit/2023-09-Gitcoin-judging/issues/727#issuecomment-1780859940)

**Comment by DevHals** [comment]
> > So basically you used the other discovered vulnerability (where the acceptedRecipient re-sets the status of the rejected milestone to Pending) as a work around solution for the described vulnerability?!
> > while re-setting the status of rejected milestones should be done by the manager only!
> 
> No. No workaround needs to be used. Recipient should redo his work the milestone requires and call `submitUpcomingMilestone` to re-submit if his submitted milestone had been rejected. This will reset the milestone status to Pending and made the redo work ready to be re-reviewed by manager.
> 
> This is not a vulnerability and is the intended workflow.

@neeksec 
 then what if:
 >What if the manager didn't wish to alter the status of the upcoming rejected milestone for valid reasons (such as the current recipient is ineligible to receive this specific milestone) but he wants to distribute the next milestone that comes after the rejected one for this recipient? he will not be able to do so since the distribution is blocked by the rejected one (and the pointer will not be updated unless this rejected one is executed)!

**Comment by neeksec** [comment]
> he wants to distribute the next milestone that comes after the rejected one for this recipient

He will not see the next milestone that comes after the rejected one because the recipient cannot submit it unless the current milestone is accepted and distributed.
He will not want to distribute the next milestone because he will never get a chance to see it.

**Comment by Evert0x** [comment]
@neeksec you are stating that the end state of a milestone can never be `rejected`, correct? The user needs to redo up until the milestone will be accepted. This will delay every milestone than comes after it as a sequential order is enforced

**Comment by neeksec** [comment]
@Evert0x
> you are stating that the end state of a milestone can never be rejected, correct? The user needs to redo up until the milestone will be accepted

Yes, that's what I was stating.

This strategy was implemented with enforced sequential order. This is not a bug. It will be a new feature to allow user to skip milestone.

**Comment by Evert0x** [comment]
@neeksec this language indicates that pool managers should also be able to reject recipients. https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/_poc/direct-grants-simple/DirectGrantsSimpleStrategy.sol#L31-L33

Which conflicts with way you interpret the code is intended to work. 

> The milestones are set by the recipient and the pool manager can accept or reject the milestone. The pool manager can also reject the recipient.

**Comment by neeksec** [comment]
@Evert0x That is another out-of-scope strategy which has nothing to do with [RFPSimpleStrategy](https://github.com/sherlock-audit/2023-09-Gitcoin/blob/main/allo-v2/contracts/strategies/rfp-simple/RFPSimpleStrategy.sol).

This diagram of RFPSimpleStrategy shows that user re-submits the upcomingMilestone after it's rejected by manager.
https://github.com/allo-protocol/allo-v2/tree/main/contracts/strategies/rfp-simple#sequence-diagram

**Comment by Evert0x** [comment]
@neeksec thanks for linking that diagram! I can see where your argument is coming from. Although it show a forced sequential order of milestones per recipient. Not necessarily a sequential order of milestones in the system. 

Asked protocol team for clarification.

**Comment by Evert0x** [comment]
Clarified will protocol team, planning to reject escalation and keep issue state as is. 

![image](https://github.com/sherlock-audit/2023-09-Gitcoin-judging/assets/35029353/60f5065f-0b63-4d74-8c51-3f74aab8fa82)

**Comment by Evert0x** [comment]
Result:
Invalid
Unique

**Comment by sherlock-admin2** [comment]
Escalations have been resolved successfully!

Escalation status:
- [DevHals](https://github.com/sherlock-audit/2023-09-Gitcoin-judging/issues/727/#issuecomment-1763214785): rejected
